Client/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -326,8 +326,15 @@ class SearchViewController: SiteTableViewController, LoaderListener {
         for engine in quickSearchEngines {
             let engineButton = UIButton()
             engineButton.setImage(engine.image, for: [])
-            engineButton.imageView?.contentMode = .scaleAspectFit
+            
+            if engine.isCustomEngine, let hostURL = fetchEngineHost(from: engine.searchTemplate) {
+                engineButton.imageView?.loadFavicon(for: hostURL, fallbackMonogramCharacter: engine.displayName.first)
```

Comment:
Does setting the image for all buttons cause any weirdness with custom engines? (Such as flickering between icons)